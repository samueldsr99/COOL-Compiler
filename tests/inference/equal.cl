class C {};

class A inherits C {};

class B inherits C {};

class Main {
    func(a: AUTO_TYPE): AUTO_TYPE {
        a = new C
    };
};

