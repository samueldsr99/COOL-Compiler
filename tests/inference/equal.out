\__ProgramNode [<class> ... <class>]
  \__ClassDeclarationNode: class C  { <feature> ... <feature> }

  \__ClassDeclarationNode: class A : C { <feature> ... <feature> }

  \__ClassDeclarationNode: class B : C { <feature> ... <feature> }

  \__ClassDeclarationNode: class Main  { <feature> ... <feature> }
    \__FuncDeclarationNode: func(a: C) : Bool -> <body>
      \__<expr> EqualNode <expr>
        \__VariableNode: a
        \__InstantiateNode: new C()
